### PR TITLE
issue 535.1: ensure recovery success if link recovery starts right at the previous marker

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

File: arangod/RestServer/FlushFeature.cpp [hunk 1]
```diff
@@ -518,9 +518,14 @@ class RocksDBRecoveryHelper final: public arangodb::RocksDBRecoveryHelper {
 };
 
 void registerRecoveryHelper() {
+  static bool done(false);
   static const MMFilesRecoveryHelper mmfilesHelper;
   static const RocksDBRecoveryHelper rocksDBHelper;
 
+  if (done) {
```
Comment (graetzer) [inline arangod/RestServer/FlushFeature.cpp]: how is this possible, are features prepared more than once?

File: arangod/IResearch/IResearchRocksDBRecoveryHelper.h [hunk 2]
```diff
@@ -82,14 +82,13 @@ class IResearchRocksDBRecoveryHelper final : public RocksDBRecoveryHelper {
  private:
   void handleDeleteCF(uint32_t column_family_id, const rocksdb::Slice& key);
 
- private:
   std::set<IndexId> _recoveredIndexes;  // set of already recovered indexes
-  DatabaseFeature* _dbFeature;
-  RocksDBEngine* _engine;
-  uint32_t _documentCF;
+  DatabaseFeature* _dbFeature{};
```
Comment (graetzer) [inline arangod/IResearch/IResearchRocksDBRecoveryHelper.h]: is this initalizting members to nullptr?

File: arangod/RestServer/FlushFeature.cpp (hunk 1 above)
Comment (vasiliy-arangodb) [reply arangod/RestServer/FlushFeature.cpp]: in tests they are, that is how I stumbled onto this

**Comment by vasiliy-arangodb** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/4154/

**Comment by gnusi** [comment]
https://jenkins.arangodb.biz/job/arangodb-matrix-pr-windows-onlycatch/2158/

**Pull request merged**